### PR TITLE
Enforce silent downcasting deprecation

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]
A reminder that we'll need to deprecate `future.no_silent_downcasting`. Could you open a follow up issue?

**Comment by mroeschke** [comment]
Thanks @phofl

**Comment by phofl** [comment]
https://github.com/pandas-dev/pandas/issues/57403

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -106,6 +106,7 @@ Removal of prior version deprecations/changes
 - :func:`read_excel`, :func:`read_json`, :func:`read_html`, and :func:`read_xml` no longer accept raw string or byte representation of the data. That type of data must be wrapped in a :py:class:`StringIO` or :py:class:`BytesIO` (:issue:`53767`)
 - All arguments except the first ``path``-like argument in IO writers are now keyword only (:issue:`54229`)
 - Changed the default value of ``observed`` in :meth:`DataFrame.groupby` and :meth:`Series.groupby` to ``True`` (:issue:`51811`)
+- Enforced silent-downcasting deprecation for :ref:`all relevant methods <whatsnew_220.silent_downcasting>` (:issue:`54710`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v3.0.0.rst:109]: looks like the option still exists in config_init and it is set in 3 tests

Comment (mroeschke) [reply doc/source/whatsnew/v3.0.0.rst:109]: The option will need to be deprecated in a follow up, but it can be removed in those tests

**Pull request merged**